res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+      DJCi500.slicerTimer[deck] = true;
+      var timer_ms = (1-passedTime)*60.0/engine.getValue(group, "bpm")*1000;
+
+      //quality of life fix for not-precise hands or beatgrid
+      // also good fix for really small timer_ms values.
+      if ( (passedTime >= 0.8) &&
+        //this is because while looping doing this thing on beat 8 break the flow.
+        ((!loopEnabled) || (DJCi500.slicerBeatsPassed[deck] % domain) !== (domain-1)) ) {
+        timer_ms += 60.0/engine.getValue(group, "bpm")*1000;
+      }
+
+      engine.beginTimer( timer_ms,
+        //"DJCi500.slicerTimerCallback("+group+")", true);
+        function() {
+          //need to do this otherwise loop does not work on beat 8 because of slip.
+          if ((engine.getValue(group, "loop_enabled") === true)){
+            //on the wiki it says it returns an integer, but I tested and instead seems a Real value:
+            // But it does not work cuz the value does not relate to beat. they are samples.
+            //var endLoop = engine.getValue(group, "loop_end_position");
+            engine.setValue(group, "reloop_toggle", true); //false
+            engine.setValue(group, "slip_enabled", false);
+            //Aleatory behavior, probably because the slip does not always have completed before "returning"
+            //so I need to introduce a timer waiting the slip function to be completely resolved
+            engine.beginTimer( 2, function () {
+              var bpm_file = engine.getValue(group, "file_bpm"),
+                playposition = engine.getValue(group, "playposition"),
+                duration = engine.getValue(group, "duration");
+              /*
+              if (Math.floor((playposition * duration) * (bpm_file / 60.0)) > endLoop) {
+                engine.setValue(group, "beatjump", -8);
+              }*/
+              engine.setValue(group, "reloop_toggle", true);},
+              true);
+          }
+          else {
+            engine.setValue(group, "slip_enabled", false);
+          }
+          DJCi500.slicerTimer[deck] = false;
+          DJCi500.slicerButton[deck] = -1;},
+        true);
+    }
+
+    engine.setValue(group, "slip_enabled", true);
+
+    //Because of Mixxx beatjump implementation, we need to deactivate the loop before jumping
+    // also there is no "lopp_deactivate" and loop_activate false does not work.
+    if (loopEnabled) {
+      engine.setValue(group, "reloop_toggle", true);
+    }
+    engine.setValue(group, "beatjump", beatsToJump);
+    //This sadly does not work.
+    //engine.setValue(group, "loop_move", -beatsToJump);
+    if (loopEnabled){
+      engine.setValue(group, "reloop_toggle", true);
+    }
+    midi.sendShortMsg((0x96+(deck % 2)), 0x20+index, 0x62);
+  } //if value
+};
+
+//this below is connected to beat_active
+DJCi500.slicerBeatActive = function(value, group, control) {
+  // This slicer implementation will work for constant beatgrids only!
+  var deck = parseInt(group.charAt(8)) - 1;

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions) and rename `deck` to `deckIndex` when its 0-based instead of 1-based.
```suggestion
  var deck = script.deckFromGroup(deckData.currentDeck) - 1;
```